File: Sources/Ably/ARTAuthDetails.m
#import "ARTAuthDetails.h"

@implementation ARTAuthDetails

- (instancetype)initWithToken:(NSString *)token {
    if (self = [super init]) {
        _accessToken = token;
    }
    return self;
}

- (NSString *)description {
    return [NSString stringWithFormat:@"%@ - \n\t accessToken: %@; \n", [super description], self.accessToken];
}

- (id)copyWithZone:(NSZone *)zone {
    ARTAuthDetails *authDetails = [[[self class] allocWithZone:zone] init];
    authDetails.accessToken = self.accessToken;
    return authDetails;
}

@end

File: Sources/AblySwift/ARTAuthDetails.swift
import Foundation

// swift-migration: original location ARTAuthDetails.h, line 8
/// Contains the token string used to authenticate a client with Ably.
public class ARTAuthDetails: NSObject, NSCopying {
    
    // swift-migration: original location ARTAuthDetails.h, line 13
    /// The authentication token string.
    public var accessToken: String
    
    // swift-migration: original location ARTAuthDetails.h, line 16
    /// :nodoc:
    public init(token: String) {
        self.accessToken = token
        super.init()
    }
    
    // swift-migration: original location ARTAuthDetails.m, line 12
    public override var description: String {
        return "\(super.description) - \n\t accessToken: \(self.accessToken); \n"
    }
    
    // swift-migration: original location ARTAuthDetails.m, line 16
    public func copy(with zone: NSZone? = nil) -> Any {
        let authDetails = ARTAuthDetails(token: self.accessToken)
        return authDetails
    }
}